src/chkreg/chkreg.c

```diff
@@ -0,0 +1,447 @@
+    {
+        g_pIdStorageUMDConfig[i] = 0;
+    }
+
+    // 0x00000318 - 0x00000334
+    for (i = 0; i < sizeof g_ConsoleIdCertificate; i++)
+    {
+        ((u8 *)&g_ConsoleIdCertificate)[i] = 0;
+    }
+
+    g_UMDRegionCodeInfoPostIndex = 0; // 0x00000348
+    g_isUMDRegionCodesObtained = SCE_FALSE;
+    g_isConsoleIdCertificateObtained = SCE_FALSE;
+
+    /* Only stop the module when it is not currently used by other modules. */
+
+    status = sceKernelWaitSema(g_semaId, 1, &timeout); // 0x0x0000035C
+    if (status == SCE_ERROR_OK) // 0x0x00000364
+    {
+        /* Module no longer in use. Stop it. */
+        sceKernelDeleteSema(g_semaId); // 0x00000380
+
+        return SCE_KERNEL_STOP_SUCCESS;
+    }
+
+    /* Module still in use, don't stop it yet. */
+
+    return SCE_KERNEL_STOP_FAIL;
+}
+
+// Subroutine sceChkreg_driver_54495B19 - Address 0x00000390
+s32 sceChkregCheckRegion(u32 umdMediaType, u32 regionId)
+{
+    s32 status1;
+    s32 status2;
+
+    status1 = SCE_ERROR_SEMAPHORE; // 0x000003BC & 0x000003C8
+
+    /* Allow only one access at a time. */
+    status2 = sceKernelWaitSema(g_semaId, 1, NULL); // 0x000003C4
+    if (status2 == SCE_ERROR_OK)
+    {
+        if (g_isUMDRegionCodesObtained || (status1 = _sceChkregLookupUMDRegionCodeInfo()) == SCE_ERROR_OK) // 0x000003D8 & 0x000003E8
+        {
+            status1 = _sceChkregCheckRegion(0x80000000, umdMediaType | regionId); // 0x000003F4
+        }
+
+        /* Release acquired sema resource. */
+        status2 = sceKernelSignalSema(g_semaId, 1); // 0x00000404
+        if (status2 != SCE_ERROR_OK)
+        {
+            status1 = SCE_ERROR_SEMAPHORE;
+        }
+    }
+
+    return status1;
+}
+
+// Subroutine sceChkreg_driver_59F8491D - Address 0x00000438
+s32 sceChkregGetPsCode(ScePsCode *pPsCode)
+{
+    s32 status1;
+    s32 status2;
+
+    status1 = SCE_ERROR_SEMAPHORE;
+
+    /* Allow only one access at a time. */
+    status2 = sceKernelWaitSema(g_semaId, 1, NULL);
+    if (status2 == SCE_ERROR_OK)
+    {
+        if ((g_isConsoleIdCertificateObtained || (status1 = _sceChkregLookupConsoleIdCertificate()) == SCE_ERROR_OK)
+            && (status1 = _sceChkregVerifyConsoleIdCertificate()) == SCE_ERROR_OK)
+        {
+            pPsCode->companyCode = g_ConsoleIdCertificate.consoleId.companyCode;
+            pPsCode->productCode = g_ConsoleIdCertificate.consoleId.productCode;
+            pPsCode->productSubCode = g_ConsoleIdCertificate.consoleId.productSubCode;
+            pPsCode->factoryCode = g_ConsoleIdCertificate.consoleId.factoryCode;
+        }
+
+        /* Release acquired sema resource. */
+        status2 = sceKernelSignalSema(g_semaId, 1);
+        if (status2 != SCE_ERROR_OK)
+        {
+            status1 = SCE_ERROR_SEMAPHORE;
+        }
+    }
+
+    return status1;
+}
+
+// Subroutine sceChkreg_driver_9C6E1D34 - Address 0x0000051C
+s32 sceChkreg_driver_9C6E1D34(const u8 *arg0, u8 *arg1)
```

Comment:
Even if we're not sure about this yet it could have better names like `encryptedData` and `hash`

Reply:
See comment [here](https://github.com/uofw/uofw/pull/91#discussion_r608198412) for first argument. `hash` could work for the second argument.

Reply:
ping @artart78 could you apply this renaming of arg1 please and add the comment for arg0. Also note that me too I found that arg0 must be a PSP_Header but I don't have a proof of usage neither.